src/hotspot/cpu/x86/macroAssembler_x86.cpp

```diff
@@ -2562,7 +2571,15 @@ void MacroAssembler::movptr(Register dst, Address src) {
 
 // src should NEVER be a real pointer. Use AddressLiteral for true pointers
 void MacroAssembler::movptr(Register dst, intptr_t src) {
-  LP64_ONLY(mov64(dst, src)) NOT_LP64(movl(dst, src));
+#ifdef _LP64
+  if (is_simm32(src)) {
+    movslq(dst, checked_cast<int32_t>(src));
```

Comment:
Why not just `movq`? there is no `movslq r, i` so this is kind of confusing.

Reply:
Right. There is no point in trying to fix `movslq`. Replaced with `movq`.